src/overlays/actors/ovl_En_Daiku2/z_en_daiku2.c

```diff
@@ -28,60 +57,526 @@ const ActorInit En_Daiku2_InitVars = {
+}
+
+void func_80BE65B4(EnDaiku2* this, GlobalContext* globalCtx) {
+    switch (gSaveContext.day - 1) {
+        case 0:
+            this->unk_28A = 0;
+            if (gSaveContext.weekEventReg[64] & 2) {
+                this->unk_28A = 6;
+            }
+            func_80BE6408(this, 8);
+            break;
+
+        case 1:
+            this->unk_28A = 2;
+            if (gSaveContext.weekEventReg[64] & 4) {
+                this->unk_28A = 7;
+            }
+            func_80BE6408(this, 8);
+            break;
+
+        case 2:
+            this->unk_28A = 4;
+            this->unk_264 = 1.0f;
+            func_80BE6408(this, 10);
+            break;
+    }
+
+    this->unk_264 = 1.0f;
+    if ((this->unk_278 >= ENDAIKU2_PARAMS_7F_0) && Flags_GetSwitch(globalCtx, this->unk_278)) {
+        this->unk_28A = 5;
+        if (this->unk_276 != 10) {
+            func_80BE6408(this, 10);
+        }
+    }
+
+    this->unk_288 = 0;
+    this->actionFunc = func_80BE66E4;
+}
+
+void func_80BE66E4(EnDaiku2* this, GlobalContext* globalCtx) {
+    f32 sp9C = this->skelAnime.animCurrentFrame;
+    s32 sp98 = gSaveContext.day - 1;
+    s32 i;
+    Vec3f sp88;
+    Vec3f sp7C;
+    Vec3f sp70;
+    s32 pad[2];
+    s16 temp_v0;
+
+    Math_SmoothStepToS(&this->actor.world.rot.y, this->actor.home.rot.y, 1, 0xBB8, 0x0);
+    if (sp98 != 2) {
+        if ((this->unk_278 >= ENDAIKU2_PARAMS_7F_0) && Flags_GetSwitch(globalCtx, this->unk_278)) {
+            this->unk_28A = 5;
+            if (this->unk_276 != 10) {
+                func_80BE6408(this, 10);
+            }
+        }
+    }
+
+    this->actor.textId = sTextIds[this->unk_28A];
+
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        func_80BE6B40(this, globalCtx);
+        return;
+    }
+
+    if ((this->unk_28A != 5) && (sp98 != 2) && func_80BE64C0(this, globalCtx)) {
+        func_80BE6CFC(this);
+        return;
+    }
+
+    func_800B8614(&this->actor, globalCtx, 80.0f);
+    if ((this->unk_276 == 8) && func_801378B8(&this->skelAnime, 6.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_ROCK_BROKEN);
+
+        for (i = 0; i < 10; i++) {
+            Math_Vec3f_Copy(&sp70, &this->actor.world.pos);
+
+            sp70.x += Math_SinS(this->actor.world.rot.y) * 70.0f;
+            sp70.y = sp70.y + 40.0f;
+            sp70.z += Math_CosS(this->actor.world.rot.y) * 70.0f;
+
+            sp70.x += randPlusMinusPoint5Scaled(5.0f);
+            sp70.y += randPlusMinusPoint5Scaled(5.0f);
+            sp70.z += randPlusMinusPoint5Scaled(5.0f);
+
+            sp88.y = -1.0f;
+
+            sp7C.x = (Rand_ZeroOne() - 0.5f) * 4.0f;
+            sp7C.y = (Rand_ZeroOne() * 10.0f) + 4.0f;
+            sp7C.z = (Rand_ZeroOne() - 0.5f) * 4.0f;
+
+            func_80BE7504(this, &sp70, &sp7C, &sp88, 0.01f - (Rand_ZeroFloat(1.0f) * 0.005f), 40);
+        }
+    }
+
+    temp_v0 = ABS_ALT(BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.home.rot.y));
+    if (temp_v0 > 0x2890) {
```

Comment:
Is the temp actually needed?
```suggestion
    if (ABS_ALT(BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.home.rot.y)) > 0x2890) {
```

Reply:
Yeah it's needed, I can't match it without using one.